File: 053_Quiz - abstract class/src/hr/unipu/java/AShortAnswerQuestion.java
package hr.unipu.java;

// Nasljeđivanje – eliminacija ponavljanja
public class AShortAnswerQuestion extends AbstractQuestion  {
    private String correctAnswer;   // atibut string

    // konstruktor
    public AShortAnswerQuestion(String text, String correctAnswer) {
        super(text);
        this.correctAnswer = correctAnswer;
    }

    // Ova metoda je zakomentirana jer je sada dio apstraktne klase
//    public String getText() {
//        return text;
//    }

    // konkrenta implementacija nasljeđene metode deklarirane kao apstraktna u apstraktnoj klasi
    public boolean isCorrectAnswer(String answer) {
        return correctAnswer.equalsIgnoreCase(answer);
    }

}

File: 053_Quiz - abstract class/src/hr/unipu/kotlin/AShortAnswerQuestion.kt
package hr.unipu.kotlin

// Nasljeđivanje – eliminacija ponavljanja
class AShortAnswerQuestion// konstruktor
(text: String, private val correctAnswer: String   // atibut string
) : AbstractQuestion(text) {

    // Ova metoda je zakomentirana jer je sada dio apstraktne klase
    //    public String getText() {
    //        return text;
    //    }

    // konkrenta implementacija nasljeđene metode deklarirane kao apstraktna u apstraktnoj klasi
    override fun isCorrectAnswer(answer: String): Boolean {
        return correctAnswer.equals(answer, ignoreCase = true)
    }

}
